src/controllers/activegate/reconciler/statefulset/reconciler.go

```diff
@@ -200,25 +201,49 @@ func (r *Reconciler) deleteStatefulSetIfOldLabelsAreUsed(desiredSts *appsv1.Stat
 	return false, nil
 }
 
-func (r *Reconciler) calculateCustomPropertyHash() (string, error) {
-	customProperties := r.capability.CustomProperties
-	if customProperties == nil || (customProperties.Value == "" && customProperties.ValueFrom == "") {
-		return "", nil
+func (r *Reconciler) calculateActiveGateHash() (string, error) {
+	customPropData, err := r.getCustomPropertyHash()
+	if err != nil {
+		return "", errors.WithStack(err)
 	}
 
-	data, err := r.getDataFromCustomProperty(customProperties)
+	authTokenData, err := r.getAuthTokenHash()
 	if err != nil {
 		return "", errors.WithStack(err)
 	}
 
 	hash := fnv.New32()
-	if _, err = hash.Write([]byte(data)); err != nil {
+	if _, err := hash.Write([]byte(customPropData + authTokenData)); err != nil {
 		return "", errors.WithStack(err)
 	}
 
 	return strconv.FormatUint(uint64(hash.Sum32()), 10), nil
 }
 
+func (r *Reconciler) getCustomPropertyHash() (string, error) {
+	if !r.needsCustomPropertyHash(r.capability.CustomProperties) {
+		return "", nil
+	}
+
+	customPropData, err := r.getDataFromCustomProperty(r.capability.CustomProperties)
```

Comment:
There is probably already a kubeobjects function to get secret data. The called function could be refactored to use it since it wouldn't be that much of a change to be out of scope

Reply:
implemented a new function which, takes a secret name and namespace as a parameter and returns the data from the secret, thanks for the input :)